xmplay/xmp_vgmstream.c

```diff
@@ -106,6 +107,8 @@ static STREAMFILE *xmpsf_open(XMPLAY_STREAMFILE *this, const char *const filenam
     newfile = xmpffile->Open(filename);
     if (!newfile) return NULL;
 
+	strcpy(filepath, filename);

```

Comment:
And change this to a strncpy, with a truncation in case it exceeds the length of the buffer.